pipeline/git.go

```diff
@@ -152,3 +157,32 @@ func updateAllCurrentPipelines() {
 	}
 	wg.Wait()
 }
+
+func subscribeRepoWebhook(repo *gaia.GitRepo) error {
+	token := os.Getenv("GAIA_GIT_TOKEN")
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	config := make(map[string]interface{})
+	config["url"] = "https://example.com/callback"
```

Comment:
So, unless echo is running and a request was received I can't determine where gaia is deployed at from the back-end side. Maybe the admin site can't, I'm not big on NPM. However, that leaves me with config options maybe. :/ Which further convolutes the huge configuration flags by adding `-host=example.com` or something like that.

I see the config refactoring approaching ever faster. :) What say you?

Reply:
Good point. I know from many tools (gitlab for example) that you have to provide the domain name where the application is accessible from. So I guess adding a configuration flag is fine here. 

I actually think we should do both. The admin can provide the domain name (which will be used for sure) and if it's not provided we try to determine it after the first request came in. What do you think?

Reply:
I agree that admin can provide a hostname which overwrites the one provided by the flag. I guess we can try. The first request would be a user login?